calender/calender.go

```diff
@@ -20,6 +20,55 @@ import (
 	"learningbay24.de/backend/models"
 )
 
+/*Returns all appointments the user with the user-ID has*/
+func GetAllAppointments(db *sql.DB, userId int) ([]models.Appointment, error) {
```

Comment:
Da du Heap-Allocated Objekte von `appointmentQuery.All(context.Background(), tx)` in Zeile 50 bekommst, geb einfach `[]*models.Appointment` zurück, damit die Objekte auf dem Heap nicht noch unnötig auf den Stack kommen